designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioActivity/Dtos.scala

```diff
@@ -0,0 +1,709 @@
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedScheduledExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object AutomaticUpdate extends ScenarioActivityType {
+      override def displayableName: String        = "Automatic update"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    override def values: immutable.IndexedSeq[ScenarioActivityType] = findValues
+
+  }
+
+  implicit def scenarioActivityTypeSchema: Schema[ScenarioActivityType] =
+    enumSchema[ScenarioActivityType](
+      ScenarioActivityType.values.toList,
+      _.entryName,
+    )
+
+  implicit def scenarioActivityTypeCodec: circe.Codec[ScenarioActivityType] = circe.Codec.from(
+    Decoder.decodeString.emap(str =>
+      ScenarioActivityType.withNameEither(str).left.map(_ => s"Invalid scenario action type [$str]")
+    ),
+    Encoder.encodeString.contramap(_.entryName),
+  )
+
+  implicit def scenarioActivityTypeTextCodec: Codec[String, ScenarioActivityType, CodecFormat.TextPlain] =
+    Codec.string.map(
+      Mapping.fromDecode[String, ScenarioActivityType] {
+        ScenarioActivityType.withNameOption(_) match {
+          case Some(value) => DecodeResult.Value(value)
+          case None        => DecodeResult.InvalidValue(Nil)
+        }
+      }(_.entryName)
+    )
+
+  def enumSchema[T](
+      items: List[T],
+      encoder: T => String,
+  ): Schema[T] =
+    Schema.string.validate(
+      Validator.enumeration(
+        items,
+        (i: T) => Some(encoder(i)),
+      ),
+    )
+
+  @derive(encoder, decoder, schema)
+  final case class ScenarioActivityMetadata(
+      `type`: String,
+      displayableName: String,
+      icon: String,
+      supportedActions: List[String],
+  )
+
+  object ScenarioActivityMetadata {
+
+    def from(scenarioActivityType: ScenarioActivityType): ScenarioActivityMetadata =
+      ScenarioActivityMetadata(
+        `type` = scenarioActivityType.entryName,
+        displayableName = scenarioActivityType.displayableName,
+        icon = scenarioActivityType.icon,
+        supportedActions = scenarioActivityType.supportedActions,
+      )
+
+  }
+
+  @derive(encoder, decoder, schema)
+  final case class ScenarioActivityActionMetadata(
+      id: String,
+      displayableName: String,
+      icon: String,
+  )
+
+  @derive(encoder, decoder, schema)
+  final case class ScenarioActivities(activities: List[ScenarioActivity])
+
+  implicit val configuration: Configuration =
```

Comment:
Can we be more precise about this configuration name because now I cannot easily find what it is used for?

Reply:
now inside the codec derivation in companion object